x-pack/elastic-agent/pkg/basecmd/version/cmd.go

```diff
@@ -5,32 +5,95 @@
 package version
 
 import (
+	"context"
 	"fmt"
 
 	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
 
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/control/client"
 	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/cli"
 	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/release"
 )
 
+// Output returns the output when `--yaml` is used.
+type Output struct {
+	Binary *release.VersionInfo `yaml:"binary"`
+	Daemon *release.VersionInfo `yaml:"daemon,omitempty"`
+}
+
 // NewCommandWithArgs returns a new version command.
 func NewCommandWithArgs(streams *cli.IOStreams) *cobra.Command {
-	return &cobra.Command{
+	cmd := &cobra.Command{
 		Use:   "version",
 		Short: "Display the version of the elastic-agent.",
-		Run: func(_ *cobra.Command, _ []string) {
-			version := release.Version()
-			if release.Snapshot() {
-				version = version + "-SNAPSHOT"
+		Run: func(cmd *cobra.Command, _ []string) {
+			var daemon *release.VersionInfo
+			var daemonError error
+
+			binary := release.Info()
+			binaryOnly, _ := cmd.Flags().GetBool("binary-only")
+			if !binaryOnly {
+				c := client.New()
+				daemonError = c.Start(context.Background())
+				if daemonError == nil {
+					defer c.Stop()
```

Comment:
i think what stop does and what i think it does are two different things

Reply:
Same as above.